projects/igniteui-angular/src/lib/icon/icon.service.ts

```diff
@@ -133,13 +138,36 @@ export class IgxIconService {
 
     /** @hidden @internal */
     public setRefsByTheme(theme: IgxTheme) {
-        if (this.theme !== theme) {
-            this.theme = theme;
+        for (const { alias, target } of iconReferences) {
+            const external = this._iconRefs.get(alias.family)?.get(alias.name)?.external;
 
-            for (const { alias, target } of iconReferences) {
-                const icon = target.get(theme) ?? target.get('default')!;
-                this.addIconRef(alias.name, alias.family, icon);
-            }
+            const _ref = this._iconRefs.get('default')?.get(alias.name) ?? {};
+            const _target = target.get(theme) ?? target.get('default')!;
+
+            const icon = target.get(theme) ?? target.get('default')!;
+            const overwrite = !external && !(JSON.stringify(_ref) === JSON.stringify(_target));
+
+            this._setIconRef(
+                alias.name,
+                alias.family,
+                icon,
+                overwrite
+            );
+        }
+    }
+
+    public _setIconRef(name: string, family: string, icon: IconMeta, overwrite = false) {
```

Comment:
Ehm, public?

Reply:
Also by the looks of it, this is more/less the same as the public `setIconRef` and perhaps it can simply call in this method with the modified as external meta + overwrite by default?